File: linux/runner/custom_plugin_registrant.cc
#include "custom_plugin_registrant.h"
#include "plugins/frame_capture_linux/frame_capture_plugin.h"

void register_custom_plugins(FlPluginRegistry* registry) {
  // Register FrameCapturePlugin
  g_autoptr(FlPluginRegistrar) frame_capture_registrar =
      fl_plugin_registry_get_registrar_for_plugin(registry, "FrameCapturePlugin");
  frame_capture_plugin_register_with_registrar(frame_capture_registrar);
}

File: macos/Runner/CustomPluginRegistrant.swift
import Cocoa
import FlutterMacOS

func RegisterCustomPlugins(registry: FlutterPluginRegistry) {
  let registrar = registry.registrar(forPlugin: "FrameCapturePlugin")
  FrameCapturePlugin.register(with: registrar)
  print("✅ CustomPluginRegistrant: FrameCapturePlugin registered")
}